src/app/notes/services/notes-preparer.service.ts

```diff
@@ -18,6 +18,11 @@ export class NotesPreparerService {
         return notes.filter(n => n.content !== undefined)
             .filter(n => n.content !== null)
             .filter(n => n.content.length !== 0)
+            .filter(n => {
+                let note = {...n};
+                let trimmedContentLength = note.content.trim().length;
```

Comment:
why not just one filter loop? 
`.filtern(n => {
    const noteContent = safe(() => n.content) || '';
    return noteContent.trim().length !== 0;
})`